package.json

```diff
@@ -35,6 +35,7 @@
     "ts-jest": "^27.0.7",
     "tslib": "^2.3.1",
     "typedoc": "^0.22.10",
+    "typedoc-plugin-missing-exports": "^0.22.6",
```

Comment:
I don't think it's worth the while to add this dependency to address a warning. Simply try and weed out those that should have been actually defined, and which I pointed out to.

Reply:
but what do you mean exactly? Cause I did add extra typedocs on the classes that you pointed out.
The issue here is that these classes are not exported - so there will be no docs generated for them and the warning is to remind so.

Reply:
Then we'll just pay no attention to those warnings. Seeing a warning (or not) does not warrant adding a new dependency, I would say. I think another possible option is simply not linking them.

Reply:
As far as I understand it, this is the normal behavior of typedoc, can't see what you mean that we can eliminate these warnings by 'not linking them' ?!

As, according to `typeDoc`, if one package includes and references type from other package, link is currently not being created but logged as warning. (e.g. [Extent](https://ideditor.github.io/id-sdk/classes/_id_sdk_extent.Extent.html) constructor, Optional max: Vec2 paramater, notice Vec2 does not link to Vector/Vec2). Explained https://github.com/TypeStrong/typedoc/issues/1739

Citing `typedoc`:

> TypeDoc 0.20 switched from documenting each file individually to documenting based on entry points. TypeDoc looks at each provided entry point and documents all exports from that entry point.
> 
> For libraries which export their full exposed API, this works well, but some packages are extremely resistant to exporting everything. 

This can be resolved by:

- using this `typedoc-plugin-missing-exports` plugin, which is exactly for this kind of libs, like ours. It mainly will look for types which are referenced, but not exported, and place them into an internal namespace for that entry point ( `<internal>`).
- or by establishing entry point order and using module resolution [paths](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping).

 I didn't think this part is so impactful which is why I chose the quicker way via the plugin. :/ 

Fine to live with some warnings too, if this is your point. But, can't see what you mean by "not linking them", @JJ

Reply:
> As far as I understand it, this is the normal behavior of typedoc, can't see what you mean that we can eliminate these warnings by 'not linking them' ?!
> 

I think that by not doing something like [[Class]] typedoc will not know it's there. Let me check anyway.

> This can be resolved by:
> 
>     * using this `typedoc-plugin-missing-exports` plugin, which is exactly for this kind of libs, like ours. It mainly will look for types which are referenced, but not exported, and place them into an internal namespace for that entry point ( `<internal>`).
> 
>     * or by establishing entry point order and using module resolution [paths](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping).

This is probably better, as long as there's no new dependency involved. Ignoring these warnings also works for me.

> 
> I didn't think this part is so impactful which is why I chose the quicker way via the plugin. :/

Well, as I say, let's just ignore them, but let me see first why we have those errors and if they can be fixed.
> 
> Fine to live with some warnings too, if this is your point. But, can't see what you mean by "not linking them", @JJ

My thoughts were that these were classes linked via this [[]]. That's probably not true, but anyway...

Reply:
Alright! agreed to live with the warnings for now on....
hope `typeDoc` will get updated soon :/